sources/api/pluto/src/proxy.rs

```diff
@@ -75,7 +91,18 @@ pub(crate) fn setup_http_client(
                 input: &https_proxy,
             })?;
     }
-    let proxy = Proxy::new(intercept, proxy_uri);
+    let mut proxy = Proxy::new(intercept, proxy_uri);
+    // Parse https_proxy as URL to extract out auth information if any
+    let proxy_url = Url::parse(&https_proxy).context(UrlParseSnafu {
+        input: &https_proxy,
+    })?;
+    if !proxy_url.username().is_empty() {
+        proxy.set_authorization(Authorization::basic(
+            proxy_url.username(),
+            proxy_url.password().unwrap_or_default(),
```

Comment:
I haven't done much with [`Proxy-Authorization`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Proxy-Authorization) but my read of the header documentation and my understanding of URL username/password encoding makes me think that it's valid to provide a passphrase without a username.

Should we be inserting the header if a password is given without a username?

Reply:
I believe so. Good catch